add vault secret decoder

svenwltr opened the pull request:
Wdyt?

Replaces this noisy code: https://github.com/rebuy-de/boombot/blob/main/pkg/dal/vault/vault.go#L74-L95

See https://github.com/rebuy-de/kubot/pull/304

svenwltr commented on pkg/vaultutil/secrets.go:
```diff
@@ -0,0 +1,54 @@
+}
+
+type OIDCSecrets struct {
+	ClientID     string `vault:"oidc-client-id"`
```
The original struct in cluster-config uses `_` (ie `oidc_client_id`). I want to have the same separator everywhere, but I do not care which one.

der-eismann replied:
I used dashes for boombot already, no hard feeling as well.

svenwltr committed:
add vault secret decoder

svenwltr commented:
I updated the decoder and am using the one that I already tested in keydb-inspector (one which actually works in production). Also I removed the struct presets, because I think they do not fit here. We can add them later, if we think it makes sense.

svenwltr commented:
bump
